### ISSUE TITLE
V3: Background Studio Updates

### ISSUE THREAD

**Issue by Eleniatari** [Whisparr/Whisparr#179]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Right now studios will only update if monitored. Depending on settings this could cause multiple unintended releases to be downloaded.

### Describe the solution you'd like

Allow unmonitored scenes of selected studios to be updated and added to whisparr's metadata list and have them populate in an unmonitored state.

### Describe alternatives you've considered

Have the ability to toggle if "new" scenes that are found and added are added in a monitored or unmonitored state.

Above but on a site-by-site basis.

**Comment by sampulsar** [comment]
SyncPerformerItems needs to be altered first as it only monitors a performers scene if it currently id not already added. 

Once that is done SyncStudioItems can be altered to add new scenes with the monitor status of the studio passed.

You are also more likely to hit an issue with the client if you have a large number of scenes/movies, 300,000+, as the client reads all of these that could have a timeout issue reading/processing all of the data for the client.

**Comment by Eleniatari** [comment]
That raises a new issue then. The ability to delete. I'm half way there just from regular use.
156,711 scenes.

**Comment by plz12345** [comment]
Your assessment of "regular use" is a bit crazy, IMO.  I don't personally envision this app as a library manager, Stash is better for that.  Primary function is download automation, not becoming a self-hosted Stash/TPDB with a downloader bolted on.

**Comment by Eleniatari** [comment]
How so, if you refresh Evil Angel for example, it pulls metadata for 20k scenes. That is a hell of a lot of scenes. Any studio that is a 'main studio' pulls all the meta data for each sub studio. It's not hard if you have 3-4 popular studios to get an obscene number of scenes.

I can't delete anything so I'm stuck with whatever is pulled.

**Comment by plz12345** [comment]
I think that's why some power users use PostgreSQL instead of SQLite.

**Comment by Eleniatari** [comment]
MetArt is 16k
Wow is 5k
Naughty America is 15k

So it's super easy if you don't realize what it does to get too many scenes that you can't control it all.

**Comment by plz12345** [comment]
More simplistic solution here, I think.  2-line code edit:

`RefreshStudioService.cs` has this function `private void SyncStudioItems(Studio studio)`

It has a huge `if` statement where it basically does nothing scene related unless the studio is monitored.  I propse two things, which I tested already

1. Remove the `if` statement
2. Set `Monitored` on newly added scenes to be equal to whatever the setting on the Studio level is.  So if Studio is unmonitored, it will bring in all scenes as unmonitored, and vice versa.  Existing scenes are untouched by this function

The one concern I'd have here is that there is no database concept of "season" in v3, because it's not based on Sonarr.  So the concept of un-monitoring specific years of scenes isn't a thing and that may be hard for users to grasp since the Studio Detail page is grouping by year for display purposes only.  That can be worked around, but it will have/keep the behavior of all newly added scenes, regardless of release date, but rather new to StashDB, will inherit the studio's monitored setting.

**Comment by sampulsar** [comment]
> grasp

There is a bigger issue as this would break how a studio or performer can be monitored. A scene is part of a studio and it also has performers. Both studios and performers can be refreshed, to add monitored scenes.

1. If a studio is not monitored but a performer is monitored should it monitored?
2. Should it depend on what adds the scene the performer refresh or the studio refresh?

I don't believe that this is intended to replicate stashDB. If you want to see all the scenes in a studio there is a link to stashDB.  

The alternative is more reasonable as its more likely to have a uses case as you are interested in monitoring a subset of a studio scene.

Are you after just "new" scenes to added?
By "new" do you mean new to stashDB, date added to stash, or new after a point in time, the release date? (Ignore scenes before 2020 etc)

**Comment by Qstick** [comment]
Right, dropping the if in RefreshStudioService.cs based on the studio being monitored is not what we want here. 

The studios view is just to show you what scenes are in the library, not every scene in Stash.

**Comment by Eleniatari** [comment]
One thing, even if I have all the scenes from a studio I still have to get the entire Studio into whsiaprr to link the scenes and monitor for new. As far as I can tell there is no flag for 'New to Stash' scenes unless you already have a full list of scenes. Since stash methodology is not the same as TPDB it's very hard to determine 'new'.

Second, if a performer is monitored and a studio is not it should follow the performer subset type, as that would be a data filter of a larger dataset.

**Comment by Eleniatari** [comment]
Some studios also never fully download either. Tried on two clean installs of whispar and the same studio does not pull 20+ episodes.

**Comment by plz12345** [comment]
> Right, dropping the if in RefreshStudioService.cs based on the studio being monitored is not what we want here. 
> 
> The studios view is just to show you what scenes are in the library, not every scene in Stash.

My proposal was showing all scenes there because it would mirror behavior from v2. 

When a Performer list in v2 added a site, that would also show all scenes. I know the intent isn't to (fully) mirror StashDB, but I think there is no harm in showing the scenes under a studio and respecting the monitor status of the studio when it refreshes to initially import them.

**Comment by Eleniatari** [comment]
This circles back to the the reduced performance when you have many scenes present monitored and unmonitored. In both page loading and manual import searching.

Without a way to filter data down I think overtime it would just cause performance degradation if you just keep adding more studios.

**Comment by Qstick** [comment]
Yes, it would. 

The intent wasn't to mirror v2, or else we would have just made changes to v2.

**Comment by plz12345** [comment]
> Yes, it would.
> 
> The intent wasn't to mirror v2, or else we would have just made changes to v2.

OK, so with that noted, should this request be closed/cancelled as "won't implement"?  While I'd love to keep parity on this part with V2, it doesn't seem like you want to go there.

**Comment by Eleniatari** [comment]
Not feasible
